elastic/kibana
# [Synthetics] Add TLS Certificate expiry alert

## Summary

Add TLS Certificate expiry alert !!

Added a toggle for TLS alert notification in monitor config for HTTP, TCP, ICMP !!

Will setup a default TLS rule which will go through all the monitors on which tls alert is enabled.

<img width="1723" alt="image" src="https://github.com/elastic/kibana/assets/3505601/8224888c-c09f-419b-9964-4e7d79218028">


### Editing
Rule can be edited from top menu

<img width="1717" alt="image" src="https://github.com/elastic/kibana/assets/3505601/c80349ff-3603-4650-a1c3-8b391f6bbd4b">


### Testing

1. Add few HTTP monitors and saved them with default tls alert on
2. You can use expired.badssl.com url for expired certificate
3. Make sure alert is being fired on default connectors set in settings
4. You can also test migration by going to 8.8 branch, by creating few http monitors there and then switching back to check if migrations is applied.
